### PR TITLE
[INTEGRATION][Flink] Prepare baseline for Flink integration which includes local Flink/Kafka environment

### PR DESCRIPTION
**Pull request by mr-yusupov**
Signed-off-by: mr-yusupov <ziyoiddin.yusupov@gmail.com>

### Problem
In order to test real integration with Flink, it is good to have local environment for Flink in connection with Kafka/SchemaRegistry.

Closes: #494 

### Solution

To reflect most common scenario for Flink applications - simple stateful Kafka to Kafka Flink application that uses Avro for serizalization.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/flink/build.gradle [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            artifactId = 'openlineage-flink'
+
+            from components.java
+
+            artifact sourceJar
+            artifact javadocJar
+
+            pom {
+                name = 'openlineage-flink'
+                description = 'Java library for OpenLineage'
```
Comment (wslulciuc) [inline integration/flink/build.gradle:210]: _Minor_: I would update to:

```
OpenLineage integration with Flink
```

File: integration/flink/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Installation
+
+Maven:
+
+```xml
+<dependency>
+    <groupId>io.openlineage</groupId>
+    <artifactId>openlineage-flink</artifactId>
+    <version>0.1.0</version>
```
Comment (wslulciuc) [inline integration/flink/README.md:13]: You'll want to align the version of the `openlineage-flink` lib with the version of the other integrations.

File: integration/flink/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# OpenLineage Flink Agent
```
Comment (wslulciuc) [inline integration/flink/README.md:1]: Will this be an agent? If not, you can just title the proj: 

```
OpenLineage Flink
```

File: integration/flink/flink-stateful-application/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# OpenLineage - Flink sample stateful application
```
Comment (wslulciuc) [inline integration/flink/flink-stateful-application/README.md:1]: _Minor_: We can make this a bit more clear that this is an flink example. I would move the example code under `examples/`:

```
integration/
   examples/
     stateful/
```

 As a reference for the directory structure, see [`proxy/examples/kafka`](https://github.com/OpenLineage/OpenLineage/tree/main/proxy/examples/kafka)

File: integration/flink/settings.gradle [hunk 5]
```diff
@@ -0,0 +1,4 @@
+rootProject.name = 'openlineage-flink'
+include 'flink-stateful-application'
```
Comment (wslulciuc) [inline integration/flink/settings.gradle:2]: Do we want to include the example flink app when building the integration?

**Comment by mobuchowski** [comment]
Might want to keep this on some branch until we have something working.

File: integration/flink/flink-stateful-application/src/test/groovy/io/openlineage/kafka/SimpleKafkaProducer.groovy [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.StringSerializer
+
+class SimpleKafkaProducer implements AutoCloseable {
+    private final KafkaProducer<String, SpecificRecord> producer
+
+    SimpleKafkaProducer(Kafka kafka) {
+        final Properties properties = new Properties()
+        properties.put('bootstrap.servers', kafka.bootstrapServers)
+        properties.put('aks', 'all')
```
Comment (pawel-big-lebowski) [inline integration/flink/flink-stateful-application/src/test/groovy/io/openlineage/kafka/SimpleKafkaProducer.groovy:16]: Did you mean `acks`?

**Comment by mr-yusupov** [comment]
image

**Pull request closed without merging**